src/uu/mcookie/src/mcookie.rs

```diff
@@ -31,10 +33,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(|v| v.as_str())
         .collect();
 
-    // TODO: Parse max size from human-readable strings (KiB, MiB, GiB etc.)
-    let max_size = matches
-        .get_one::<String>(options::MAX_SIZE)
-        .map(|v| v.parse::<u64>().expect("Failed to parse max-size value"));
+    let max_size = if let Some(size_str) = matches.get_one::<String>(options::MAX_SIZE) {
+        match Size::parse(size_str) {
+            Ok(size) => Some(size.size_bytes()),
+            Err(_) => {
+                eprintln!("Failed to parse max-size value");
+                std::process::exit(1);
```

Comment:
While this works fine, we usually use `USimpleError` from `uucore` for errors. Something like:
```rust
return Err(USimpleError::new(1, "Failed to parse max-size value"));
```